aggregator/aggregator.go

```diff
@@ -369,48 +363,45 @@ func (agg *aggregator) updateShardsWithLock(newPlacement services.Placement) err
 	}
 
 	var (
-		newShards = instance.Shards()
-		incoming  []*aggregatorShard
-		closing   []*aggregatorShard
+		newShardSet = instance.Shards()
+		incoming    []*aggregatorShard
+		closing     []*aggregatorShard
 	)
 	for _, shard := range agg.shards {
 		if shard == nil {
 			continue
 		}
-		if !newShards.Contains(shard.ID()) {
+		if !newShardSet.Contains(shard.ID()) {
 			closing = append(closing, shard)
 		}
 	}
-	// NB(xichen): shard ids are guaranteed to be sorted in ascending order.
-	newShardIDs := newShards.AllIDs()
-	if len(newShardIDs) > 0 {
-		newShardLen := newShardIDs[len(newShardIDs)-1] + 1
+
+	// NB(xichen): shards are guaranteed to be sorted by their ids in ascending order.
+	var (
+		newShards   = newShardSet.All()
+		newShardIDs []uint32
+	)
+	if numShards := len(newShards); numShards > 0 {
+		newShardIDs = make([]uint32, 0, numShards)
+		newShardLen := newShards[numShards-1].ID() + 1
```

Comment:
Is the `+1` here just so we can do `incoming[shardID]` instead of `incoming[shardID-1]`? Do the shard ID's begin at 1 then instead of 0?

Reply:
This is because the shard ids start at 0, so the length needs to be 1 larger than the max id.

Reply:
Gotcha, yea disregard this comment, was too early I guess :)

Reply:
nit: Would be clearer to do 
`
maxShardID := newShards[numShards-1].ID()

incoming = make([]*aggregatorShard, maxShardID + 1)
`

Reply:
Sure can do.